### PR TITLE
Extending Windows chef-client compatibility 

### PR DESCRIPTION
**Pull request by sam1el**
### Description

This will allow habitat to be installed with chef-client 12+ in support of upgrading customers to effortless as well as deploying habitat in general. Also, added some idempotence to the install resource. 

### Issues Resolved

The install resource was not 100% idempotent on windows. that is now fixed. Added custom unzip function to allow older chef-client versions to decompress the zip archive on windows. 

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by collinmcneese** [comment]
👍

File: resources/install.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 resource_name :hab_install
 
-property :name, String, default: ''
+property :name, String, default: '' # ~FC108 This allows for bare names like hab_install
```
Comment (tas50) [inline resources/install.rb]: We shouldn't be testing using foodcritic anymore

File: resources/install.rb [hunk 2]
```diff
@@ -175,7 +201,7 @@ def hab_path
   end
 
   def hab_command
-    cmd = if node['kernel']['release'].to_i < 3
+    cmd = if node['kernel']['release'].to_f < 3.0
```
Comment (tas50) [inline resources/install.rb]: did this sneak back in or was it necessary?

File: resources/install.rb [hunk 3]
```diff
@@ -124,11 +136,25 @@
       source download
     end
 
-    archive_file "#{package_name}.zip" do
-      path zipfile
-      destination "#{Chef::Config[:file_cache_path]}/habitat"
-      overwrite true
-      action :extract
+    if Chef::VERSION < 15
+      ruby_block 'name' do
```
Comment (tas50) [inline resources/install.rb]: This should probably have a real name

File: resources/install.rb [hunk 4]
```diff
@@ -50,26 +50,38 @@
...
+          require 'zip'
+          Zip::File.open(zipfile) do |zip_file|
+            zip_file.each do |f|
+              fpath = "#{Chef::Config[:file_cache_path]}/habitat/" + f.name
+              zip_file.extract(f, fpath) # unless ::File.exist?(fpath)
+            end
+          end
+        end
+        action :run
+        not_if { Dir.exist?('c:\habitat') }
```
Comment (tas50) [inline resources/install.rb]:
```suggestion
        not_if { ::Dir.exist?('c:\habitat') }
```

File: resources/install.rb [hunk 5]
```diff
@@ -124,11 +136,25 @@
...
-    archive_file "#{package_name}.zip" do
-      path zipfile
-      destination "#{Chef::Config[:file_cache_path]}/habitat"
-      overwrite true
-      action :extract
+    if Chef::VERSION < 15
+      ruby_block 'name' do
+        block do
+          require 'zip'
+          Zip::File.open(zipfile) do |zip_file|
```
Comment (tas50) [inline resources/install.rb]: should we have a not_if on this ruby_block as well?

Comment (collinmcneese) [reply resources/install.rb]: This one didn't have a guard because it is in the upgrade action, for upgrade we want it to run no matter what

**Pull request merged**